### PR TITLE
Fix `AutoInject` code formatting in the `TodoTemplate` project (#2321)

### LINKED ISSUE #2321: Incorrect `AutoInject` code formatting in the `TodoTemplate` project
**Correct format :**

- Server & Shared:
[AutoInject] private ICustomerService customerService;

- Client:
[AutoInject] private ICustomerService customerService = default!;

### PR DESCRIPTION
**Pull request by sajadkardel**
this closes #2321

### CODE REVIEW

File: src/Templates/TodoTemplate/Bit.TodoTemplate/Web/Shared/TodoTemplateErrorBoundary.razor.cs [hunk 1]
```diff
@@ -5,9 +5,9 @@ public partial class TodoTemplateErrorBoundary
 {
     private bool ShowException { get; set; }
 
-    [AutoInject] private IExceptionHandler ExceptionHandler { get; set; } = default!;
+    [AutoInject] private IExceptionHandler ExceptionHandler = default!;
```
Comment (ysmoradi) [inline src/Templates/TodoTemplate/Bit.TodoTemplate/Web/Shared/TodoTemplateErrorBoundary.razor.cs:8]: Use field name convention for all fields @sajadkardel

**Review by ysmoradi — CHANGES_REQUESTED** [review]

File: src/Templates/TodoTemplate/Bit.TodoTemplate/Web/Shared/TodoTemplateErrorBoundary.razor.cs (hunk 1 above)
Comment (sajadkardel) [reply src/Templates/TodoTemplate/Bit.TodoTemplate/Web/Shared/TodoTemplateErrorBoundary.razor.cs:8]: `customerService` or `_customerService` ?

Comment (ysmoradi) [reply src/Templates/TodoTemplate/Bit.TodoTemplate/Web/Shared/TodoTemplateErrorBoundary.razor.cs:8]: I'd prefer `customerService`

File: src/Templates/TodoTemplate/Bit.TodoTemplate/Web/Components/NavMenu.razor.cs [hunk 2]
```diff
@@ -59,41 +59,41 @@ public bool IsMenuOpen
         {
             if (value == isMenuOpen) return;
             isMenuOpen = value;
-            _ = IsMenuOpenChanged.InvokeAsync(value);
+            _ = isMenuOpenChanged.InvokeAsync(value);
         }
     }
 
-    [Parameter] public EventCallback<bool> IsMenuOpenChanged { get; set; }
+    [Parameter] public EventCallback<bool> isMenuOpenChanged { get; set; }
```
Comment (msynk) [inline src/Templates/TodoTemplate/Bit.TodoTemplate/Web/Components/NavMenu.razor.cs:66]: this is a parameter and this change should be rolled back

Comment (ysmoradi) [inline src/Templates/TodoTemplate/Bit.TodoTemplate/Web/Components/NavMenu.razor.cs:66]: `IsMenuOpenChanged` is a property

**Review by ysmoradi — DISMISSED** [review]

File: src/Templates/TodoTemplate/Bit.TodoTemplate/Web/Components/NavMenu.razor.cs [hunk 3]
```diff
@@ -59,41 +59,41 @@ public bool IsMenuOpen
...
-    [AutoInject] private IStateService StateService { get; set; } = default!;
+    [AutoInject] private IStateService stateService = default!;
 
-    [AutoInject] private IAuthTokenProvider AuthTokenProvider { get; set; } = default!;
+    [AutoInject] private IAuthTokenProvider authTokenProvider = default!;
 
 #if BlazorServer || BlazorHybrid
-    [AutoInject] private IConfiguration Configuration { get; set; } = default!;
+    [AutoInject] private IConfiguration configuration = default!;
 #endif
```
Comment (msynk) [inline src/Templates/TodoTemplate/Bit.TodoTemplate/Web/Components/NavMenu.razor.cs:76]: plz move all the injected fields to the top of the file.

**Review by msynk — CHANGES_REQUESTED** [review]

**Comment by ysmoradi** [comment]
Could you please resolve the conflicts? @sajadkardel

File: src/Templates/TodoTemplate/Bit.TodoTemplate/Web/Components/NavMenu.razor.cs [hunk 4]
```diff
@@ -5,6 +5,17 @@ namespace TodoTemplate.App.Components;
...
+
+    [AutoInject] private IStateService stateService = default!;
+
+    [AutoInject] private IAuthTokenProvider authTokenProvider = default!;
+
+#if BlazorServer || BlazorHybrid
+    [AutoInject] private IConfiguration configuration = default!;
+#endif
+
+    private bool IsMenuOpenHasBeenSet;
```
Comment (msynk) [inline src/Templates/TodoTemplate/Bit.TodoTemplate/Web/Components/NavMenu.razor.cs:18]: you shouldn't add this field anymore

**Review by msynk — CHANGES_REQUESTED** [review]
there is still one more usage of the [Inject] attribute in the `SignOutConfirmModal.razor` which need to be replaced with [AutoInject]

**Review by msynk — APPROVED** [review]

**Pull request merged**